azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/toolkit/lib/common/utils/InstallationIdUtils.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+package com.microsoft.azure.toolkit.lib.common.utils;
+
+import org.apache.commons.lang3.StringUtils;
+
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class InstallationIdUtils {
+
+    private static final Pattern HASHED_MAC_PATTERN = Pattern.compile("[0-9a-f]{64}");
+    // Hashed mac address for iBridge device
+    private static final String INVALID_HASHED_MAC_ADDRESS = "6c9d2bc8f91b89624add29c0abeae7fb42bf539fa1cdb2e3e57cd668fa9bcead";
+
+    public static boolean isValidHashMac(String hashMac) {
+        return isValidHash(hashMac) && !isDepreciatedHashMac(hashMac);
+    }
+
+    public static String getHashMac() {
+        String mac = NetUtils.getMac();
```

Comment:
You have missed the logic of `getRawMacByCommand`
```java
public static String getMac() {
        try {
            InetAddress ip = InetAddress.getLocalHost();
            NetworkInterface network = NetworkInterface.getByInetAddress(ip);
            byte[] mac = network.getHardwareAddress();
            StringBuilder sb = new StringBuilder();
            for (int i = 0; i < mac.length; i++) {
                sb.append(String.format("%02X%s", mac[i], (i < mac.length - 1) ? "-" : ""));
            }
            return sb.toString();
            /**
             * TODO: refactor it to adopt exception case in mac.
             */
        } catch (Throwable e) {
            return "UNKNOWN_MAC";
        }
    }
```

![image](https://user-images.githubusercontent.com/17075863/120949413-0abb1300-c777-11eb-9b09-14844735db5f.png)